canal/sync.go

```diff
@@ -101,19 +102,24 @@ func (c *Canal) runSyncBinlog() error {
 			}
 		case *replication.QueryEvent:
 			// handle alert table query
-			if mb := expAlterTable.FindSubmatch(e.Query); mb != nil {
-				if len(mb[1]) == 0 {
-					mb[1] = e.Schema
+			var mb = [][]byte{}
+			// run alter table directly
+			if mb = expAlterTable.FindSubmatch(e.Query); mb == nil {
```

Comment:
Maybe you still have some misunderstanding about `pt-online-schema-change`.

Assume we want change the structure of table A:
step 1: Create a duplicate table B from A first. 
step 2: Apply alter DDL on B, 
step 3: Rename A to C, B to A(atomic, by one DDL, maybe like `rename table A to C, B to A`).

As above steps, we just focus on the rename event(QueryEvent in here) in step 3.

In general, process alter DDL **OR** rename DDL is cover all situation.